libvast/vast/sketch/sketch.hpp

```diff
@@ -0,0 +1,37 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include "vast/chunk.hpp"
+
+#include <cstddef>
+
+namespace vast::sketch {
+
+/// A type-erased sketch.
+class sketch {
+public:
+  /// Constructs a partition sketch from a flatbuffer.
+  explicit sketch(chunk_ptr flatbuffer) noexcept;
+
+  /// Checks whether the sketch fulfils a given predicate.
+  /// @param pred The predicate to check.
+  /// @returns A boolean that indicates whether *x* is in the sketch, or
+  /// std::nullopt if the query cannot be answered.
+  std::optional<bool> lookup(relational_operator op, const data& x) const noexcept;
+
+  friend size_t mem_usage(const sketch& x);
+
+private:
+  chunk_ptr flatbuffer_;
```

Comment:
@dominiklohmann would this be a use case for `flatbuffer<T>`?

Reply:
Yes, and similarly so the constructor, which should already receive a `flatbuffer<T>`.

Reply:
Does 78a5c02 look good to you in this regard?

Reply:
Yes!